### ISSUE TITLE
Support bulk upload via REST APIs

### ISSUE THREAD

**Issue by adbharadwaj** [Murali-group/GraphSpace#251]

**Comment by sandeepm96** [comment]
I would like to work on this. I had a doubt. Will there be a separate API endpoint for this or will it be implemented in the /graphs endpoint itself?

**Comment by adbharadwaj** [comment]
We plan to make a separate endpoint. Since the bulk upload will take a lot
of time (indexing will take a lot of time), we cannot make the users wait
till all the graphs have been added.

*Aditya Bharadwaj*
Graduate Research Assistant,
Department of Computer Science,
Virginia Tech
adb@vt.edu

On Thu, Apr 13, 2017 at 1:00 PM, Sandeep Mahapatra <notifications@github.com

**Comment by sandeepm96** [comment]
I read up on this [blog post](http://restlet.com/company/blog/2015/05/18/implementing-bulk-updates-within-restful-services/) that implementing separate endpoints will not be RESTful. We can also do it by having different content types or different action identifiers for the same post request. @adbharadwaj What should be our best approaach considering the above? 

Also I read up about asynchronous processing on this [blog post](http://farazdagi.com/blog/2014/rest-long-running-jobs/) which we could implement in our case. This way we will not make the users wait for bulk uploads.

**Comment by adbharadwaj** [comment]
@sandeepm96 You are right that implementing separate endpoints will not be RESTful. This [blog post](http://farazdagi.com/blog/2014/rest-long-running-jobs/) points out the correct specification for the REST APIs. This solution will require a usage of a queueing system like [Kafka](https://kafka.apache.org/intro).  This might not be an ideal start for you learn about the GraphSpace ecosystem.
